### PR TITLE
BUG: boxplot does not work when data has datetime column

### LINKED ISSUE #22799: BUG: boxplot does not work when there is a datetime64 column
Making a boxplot of a DataFrame does not seem to skip datetime64 columns, as you get the following error (it tries to calculate the mean of the datetime64 values):

```
In [4]: df = pd.DataFrame({'a': pd.date_range("2012-01-01", periods=100), 'b': np.random.randn(100), 'c': np.random.randn(100)+2})

In [6]: df.plot(kind='box')
...
TypeError: unsupported operand type(s) for +: 'Timestamp' and 'Timestamp'
```

While when this would be a string column, it would be correctly skipped.

This seems to be a regression, as at least in pandas 0.20.3 this worked.

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] closes #22799
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -160,7 +160,7 @@ Plotting
 
 -
 -
-
+- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datatime column (:issue:`22799`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datatime column (:issue:`22799`)

```

(missing new line)

Comment (charlesdong1991) [reply doc/source/whatsnew/v1.0.0.rst]: thanks, added!

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -400,9 +400,14 @@ def _compute_plot_data(self):
...
         # with ``dtype == object``
         data = data._convert(datetime=True, timedelta=True)
-        numeric_data = data.select_dtypes(
-            include=[np.number, "datetime", "datetimetz", "timedelta"]
-        )
+
+        # GH22799, exclude datatime type for boxplot
+        include_type = [np.number, "datetime", "datetimetz", "timedelta"]
+        if self._kind == "box":
+            include_type.remove("datetime")
```
Comment (TomAugspurger) [inline pandas/plotting/_matplotlib/core.py]: Does datetimetz work? Or does it need to be excluded here too?

Comment (charlesdong1991) [reply pandas/plotting/_matplotlib/core.py]: ahh, yeah, should exclude `datetime` and `timedelta`... thanks for pointing out! @TomAugspurger

File: doc/source/whatsnew/v1.0.0.rst [hunk 3]
```diff
@@ -165,6 +165,7 @@ Plotting
 
 - Bug in :meth:`Series.plot` not able to plot boolean values (:issue:`23719`)
 -
+- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datatime column (:issue:`22799`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datetime or timedelta data. These types are now automatically dropped (:issue:`22799`)
```

**Review by TomAugspurger — COMMENTED** [review]
Small comment on the release not. LGTM otherwise.

File: doc/source/whatsnew/v1.0.0.rst (hunk 3 above)
Comment (charlesdong1991) [reply doc/source/whatsnew/v1.0.0.rst]: Thanks for taking a look! Updated! @TomAugspurger

**Comment by TomAugspurger** [comment]
Thanks @charlesdong1991!

**Pull request merged**